src/pyFAI/io/integration_config.py

```diff
@@ -247,18 +328,23 @@ def normalize(config, inplace=False, do_raise=False):
             raise ValueError(txt)
         else:
             _logger.error(txt)
-
-    if version == 3:
+    if config["version"] == target_version: return config
+    if config["version"] == 3:
         _patch_v3_to_v4(config)
 
-    if version > 4:
-        _logger.error("Configuration file %d too recent. This version of pyFAI maybe too old to read this configuration", version)
+    if config["version"] == target_version: return config
+    if config["version"] == 4:
+        _patch_v4_to_v5(config)
+
+    if version > CURRENT_VERSION:
+        _logger.error("Configuration file %d too recent. This version of pyFAI maybe too old to read this configuration (max=%d)",
+                      version, CURRENT_VERSION)
 
     return config
 
 
 class ConfigurationReader(object):
-
+    "This class should be deprecated now ..."
```

Comment:
Maybe add a deprecation warning in the constructor

Reply:
Indeed, TODO